packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -0,0 +1,367 @@
+import { useState, useCallback, useMemo, forwardRef } from 'react';
+import type { FileUploadProps, BladeFile, BladeFileList } from './types';
+import { StyledFileUploadWrapper } from './StyledFileUploadWrapper';
+import { fileUploadColorTokens, getFileUploadInputHoverTokens } from './fileUploadTokens';
+import { FileUploadItem } from './FileUploadItem';
+import { isFileAccepted } from './isFileAccepted';
+import BaseBox from '~components/Box/BaseBox';
+import { Box } from '~components/Box';
+import { SelectorLabel } from '~components/Form/Selector/SelectorLabel';
+import { SelectorInput } from '~components/Form/Selector/SelectorInput';
+import { screenReaderStyles } from '~components/VisuallyHidden';
+import { FormHint, FormLabel } from '~components/Form';
+import { useFormId } from '~components/Form/useFormId';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+import { getStyledProps } from '~components/Box/styledProps';
+import { makeSize, useTheme } from '~utils';
+import { Text } from '~components/Typography';
+import type { BladeElementRef } from '~utils/types';
+import { getHintType } from '~components/Input/BaseInput/BaseInput';
+import { makeAccessible } from '~utils/makeAccessible';
+
+const _FileUpload: React.ForwardRefRenderFunction<BladeElementRef, FileUploadProps> = (
+  {
+    name,
+    accept,
+    uploadType = 'single',
+    onChange,
+    onPreview,
+    onRemove,
+    onCancel,
+    onDrop,
+    isDisabled,
+    isRequired,
+    necessityIndicator,
+    fileList,
+    testID,
+    label,
+    labelPosition = 'top',
+    accessibilityLabel,
+    validationState,
+    helpText,
+    errorText,
+    maxCount,
+    maxSize,
+    ...styledProps
+  },
+  ref,
+): React.ReactElement => {
+  const { platform } = useTheme();
+  const [selectedFiles, setSelectedFiles] = useState<BladeFileList>(fileList ?? []);
+  const [errorMessage, setErrorMessage] = useState(errorText);
+  const [internalValidationState, setInternalValidationState] = useState('none');
+  const [isActive, setIsActive] = useState(false);
+
+  const isMultiple = uploadType === 'multiple';
+  const isOneFileSelectedWithSingleUpload = !isMultiple && selectedFiles.length === 1;
+  const inputLabelPosition = platform === 'onMobile' ? 'top' : labelPosition;
+  const isLabelLeftPositioned = inputLabelPosition === 'left';
+  const willRenderHintText = isOneFileSelectedWithSingleUpload
+    ? false
+    : Boolean(helpText) || Boolean(errorMessage);
```

Comment:
We dont render helpText in mutli upload?

Reply:
We do, fixed.